hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientClusterServiceImpl.java

```diff
@@ -91,13 +90,16 @@ private MemberListSnapshot(int version, LinkedHashMap<UUID, Member> members, UUI
         }
     }
 
-    public ClientClusterServiceImpl(HazelcastClientInstanceImpl clientInstance, ILogger logger) {
+    public ClientClusterServiceImpl(ILogger logger) {
         this.logger = logger;
-        this.client = clientInstance;
     }
 
     public Cluster getCluster() {
-        return new ClientClusterProxy(this, this.client.getConnectionManager());
+        return new ClientClusterProxy(this);
+    }
+
+    public boolean isEnterprise() {
+        return this.getMemberList().size() > 0 && this.isFailoverSupported;
```

Comment:
why do you need this check? `this.getMemberList().size()`

Reply:
because we can be disconnected. we want to return false in that case

Reply:
We want Cluster.isEnterprise to return true only if we are connected to a EE cluster. If we are disconnected, it should return false. Therefore, we have this check.

Reply:
why? we were connected once and we know it was EE and will try to connect again. In the meanwhile, this API would just return the last result. The cluster type will not change until a failover happens, right?

Reply:
The migration cluster should know if target is EE or not. It may connect to the target cluster and it can be EE. Then, let's say the user shuts down the target cluster and restarts it as OS. Until we connect to target again, the isEnterprise check will pass. It is hard to make mistakes here. I think requiring connectivity in the EE check makes sense.

Reply:
Then, at least can we change the method name to indicate this? e.g. `isConnectedToEnterpriseCluster` ?

Reply:
Shouldn't it rather throw an exception?

Look how we are going to use it: We create the connection, check if it is EE, the client is disconnected for whatever reason and we fail saying we can't continue because the client is not EE (while the correct message is that the client is not connected).

Reply:
Right let me change this to throwing an exception instead.

Reply:
@ihsandemir After making the method throw, I dont think we need such a long name. Cluster.isEnterprise already enough. Only extra you have in your method is "Connected". With throwing if we are not connected, we should not need it in the name anymore.

Reply:
If you throw an exception, since it is a public API, please put it int he method signature. If the exception is added, the original name would be enough.